### PR TITLE
Maintain `Traverse` and `TraverseFilter` documents

### PR DESCRIPTION
**Pull request by hadilq**
- Create `typeclasses/traverse/README.md`
- Complete `typeclasses/traversefilter/README.md`
- Fix link in `typeclasses/functor/README.md`
- Modify `Traverse.kt` and `TraverseFilter.kt` comments

### CODE REVIEW

File: arrow-docs/docs/arrow/typeclasses/traverse/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+layout: docs-core
+title: Traverse
+permalink: /arrow/typeclasses/traverse/
+---
+
+## Traverse
+
+
```
Comment (pakoito) [inline arrow-docs/docs/arrow/typeclasses/traverse/README.md:9]: Additional empty lines

File: arrow-docs/docs/arrow/typeclasses/traverse/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+The `Traverse` typeclass is allowing to commute types from `F<G<A>>` to `G<F<A>>` over sequential execution of code.
+Traversal over a structure with an effect.
```
Comment (pakoito) [inline arrow-docs/docs/arrow/typeclasses/traverse/README.md:13]: `The main use of this is traversal over...`

File: arrow-docs/docs/arrow/typeclasses/traverse/README.md (hunk 1 above)
Comment (hadilq) [reply arrow-docs/docs/arrow/typeclasses/traverse/README.md:9]: I checked and there is exactly 4 lines in `Functor`, `Foldable`, and `Monad`, etc. So I concluded it's okey to have them in this doc too. lmkwdyt.

File: arrow-docs/docs/arrow/typeclasses/traverse/README.md (hunk 2 above)
Comment (hadilq) [reply arrow-docs/docs/arrow/typeclasses/traverse/README.md]: Thanks! See 4e5f8d68

File: arrow-docs/docs/arrow/typeclasses/traverse/README.md (hunk 1 above)
Comment (pakoito) [reply arrow-docs/docs/arrow/typeclasses/traverse/README.md:9]: weird. Okay, let's roll with it.

**Pull request merged**